src/main/java/com/app/onlinebookstore/service/impl/CategoriesServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.app.onlinebookstore.service.impl;
+
+import com.app.onlinebookstore.dto.BookDtoWithoutCategoryIds;
+import com.app.onlinebookstore.dto.CategoryDto;
+import com.app.onlinebookstore.dto.CreateCategoryRequestDto;
+import com.app.onlinebookstore.exaption.EntityNotFoundException;
+import com.app.onlinebookstore.mapper.CategoryMapper;
+import com.app.onlinebookstore.model.Category;
+import com.app.onlinebookstore.repository.CategoryRepository;
+import com.app.onlinebookstore.service.CategoryService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoriesServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category with id " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category category = categoryMapper.toModel(requestDto);
+        category.setId(id);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoriesId(Long id) {
+        return null;

```

Comment:
Did you test your endpoints in postman? I guess this method won't work correctly, because it returns null

Reply:
It works corectly.  I don't khow why